src/Stratis.FederatedPeg.Features.FederationGateway/Notifications/BlockObserver.cs

```diff
@@ -49,9 +69,21 @@ protected override void OnNextCore(ChainedHeaderBlock chainedHeaderBlock)
 
             // todo: persist the last seen block height in database
             // todo: save these deposits in local database
+            var newlyMaturedBlock = GetNewlyMaturedBlock(chainedHeaderBlock);
+            if (newlyMaturedBlock == null) return;
+
             var deposits = this.depositExtractor.ExtractDepositsFromBlock(
-                chainedHeaderBlock.Block,
-                chainedHeaderBlock.ChainedHeader.Height);
+                newlyMaturedBlock.Block,
+                newlyMaturedBlock.Height);
+        }
+
+        private ChainedHeader GetNewlyMaturedBlock(ChainedHeaderBlock latestPublishedBlock)
+        {
+            var newMaturedHeight = latestPublishedBlock.ChainedHeader.Height - (int)this.minimumDepositConfirmations;

```

Comment:
I feel like it may be better to walk down the `ChainedHeader`s to find the block that has just become mature.